pkg/destroy/azure/azure.go

```diff
@@ -0,0 +1,247 @@
+				}
+				logger.WithField("record", to.String(record.Name)).Info("deleted")
+			}
+		}
+	}
+	return nil
+}
+
+// getSharedDNSZone returns the closest parent public dns zone for privZoneName
+func getSharedDNSZone(ctx context.Context, client dns.ZonesClient, privZoneName string) (dnsZone, error) {
+	domain := privZoneName
+	parents := sets.NewString(domain)
+	for {
+		idx := strings.Index(domain, ".")
+		if idx == -1 {
+			break
+		}
+		if len(domain[idx+1:]) > 0 {
+			parents.Insert(domain[idx+1:])
+		}
+		domain = domain[idx+1:]
+	}
+
+	allPublicZones := []dnsZone{}
+	for zonesPage, err := client.List(ctx, to.Int32Ptr(100)); zonesPage.NotDone(); err = zonesPage.NextWithContext(ctx) {
+		if err != nil {
+			return dnsZone{}, err
+		}
+		for _, zone := range zonesPage.Values() {
+			if zone.ZoneType == dns.Public && parents.Has(to.String(zone.Name)) {
+				allPublicZones = append(allPublicZones, dnsZone{Name: to.String(zone.Name), ID: to.String(zone.ID), Group: groupFromID(to.String(zone.ID)), Public: true})
+				continue
+			}
+		}
+	}
+	sort.Slice(allPublicZones, func(i, j int) bool { return len(allPublicZones[i].Name) > len(allPublicZones[j].Name) })
+	if len(allPublicZones) > 0 {
+		return allPublicZones[0], nil
+	}
+	return dnsZone{}, errNoMatchingSharedZone
+}
+
+var errNoMatchingSharedZone = errors.New("no matching shared zone found")
+
+type dnsZone struct {
+	Name   string
+	ID     string
+	Group  string
+	Public bool
+}
+
+func groupFromID(id string) string {
+	return strings.Split(id, "/")[4]
+}
+
+func toRecordType(t string) dns.RecordType {
+	return dns.RecordType(strings.TrimPrefix(t, "Microsoft.Network/dnszones/"))
+}
+
+func deleteResourceGroup(client resources.GroupsGroupClient, logger logrus.FieldLogger, name string) error {
+	logger = logger.WithField("resource group", name)
+	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Minute)
+	defer cancel()
+
+	delFuture, err := client.Delete(ctx, name)
+	if err != nil {
+		return err
+	}
+
+	err = delFuture.WaitForCompletionRef(ctx, client.Client)
+	if err != nil {
+		if wasNotFound(delFuture.Response()) {
+			logger.Debug("already deleted")
+			return nil
+		}
+		return errors.Wrapf(err, "failed to delete %s", name)
+	}
+	logger.Info("deleted")
+	return nil
+}
+
+// New returns azure Uninstaller from ClusterMetadata.
+func New(logger logrus.FieldLogger, metadata *types.ClusterMetadata) (destroy.Destroyer, error) {
+	sess, err := session.GetSession()
+	if err != nil {
+		return nil, err
+	}
+
+	return &ClusterUninstaller{
+		SubscriptionID: sess.Credentials.SubscriptionID,
+		Authorizer:     sess.Authorizer,
+		InfraID:        metadata.InfraID,
+		Logger:         logger,
+	}, nil
+}
+
+func wasNotFound(resp *http.Response) bool {
+	return responseWasStatusCode(resp, http.StatusNotFound)
+}
+
+func responseWasStatusCode(resp *http.Response, statusCode int) bool {
+	if r := resp; r != nil {
+		if r.StatusCode == statusCode {
```

Comment:
Why `r`?  Can you inline as:

```go
resp != nil && resp.StatusCode == http.StatusNotFound
```

and drop these two helper functions?

Reply:
done with https://github.com/openshift/installer/pull/1771/commits/3d25470009039e20339d645c8a4454559406e914 -> https://github.com/openshift/installer/pull/1771/commits/27da87f528d10f6dd1d7678e7ec859365c1df300

kept the wasNotFound function as it is used in 2 places.